### PR TITLE
Always update package cache where we can.

### PR DESCRIPTION
**Pull request by dschaper**
**By submitting this pull request, I confirm the following (please check boxes, eg [X]) _Failure to fill the template will close your PR_:**

***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**
 
10

---
Instead of only checking for cache older than one day, always check Debian based cache, the repo's could be updated at any time. On Fedora, no need to check the cache, it's updated every time a package install is called for.

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

### CODE REVIEW

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -718,19 +717,13 @@ update_pacakge_cache() {
...
+  #Update package cache on apt based OSes. Do this every time since
+  #it's quick and packages can be updated at any time.
 
-  if [ ! "${today}" == "${timestampAsDate}" ]; then
-    #update package lists
     echo ":::"
-    echo -n "::: ${PKG_MANAGER} update has not been run today. Running now..."
+    echo -n "::: Updating local cache of available packages..."
     ${UPDATE_PKG_CACHE} &> /dev/null
     echo " done!"
```
Comment (Mcat12) [inline automated install/basic-install.sh]: Remove one indentation level

Comment (dschaper) [inline automated install/basic-install.sh]: ?

Comment (Mcat12) [inline automated install/basic-install.sh]: Since you removed an if statement, the code should move left 2 spaces.

Comment (dschaper) [inline automated install/basic-install.sh]: Got ya...

**Comment by Mcat12** [comment]
Approved

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/feedback-for-allow-defining-clients-by-their-mac-address/32324/4

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/update-package-cache-only-once-in-24h/32371/1

**Pull request merged**